app/models/course/assessment/answer/programming.rb

```diff
@@ -58,7 +58,15 @@ def assign_params(params)
 
     params[:files_attributes]&.each do |file_attributes|
       file = files.find { |f| f.id == file_attributes[:id].to_i }
-      file.content = file_attributes[:content]
+      if file.present?
```

Comment:
I don't think we want to be allowing any one who can update a programming question to be freely creating and destroying files (especially for the single file programming questions). 

There are a few fixes, perhaps you can explore and see what is best? (Either detect the question here, or enforce it in the params in the submission update service).

Reply:
I've exposed another route specifically to update answer files, and abstracted this into a separate model function. On the front-end, other questions won't be using the same route (so only file_submission questions will be using this route), and there is an additional check in the service to ensure this.

Reply:
@zhengwei143 Could you link me to the service that does the checking of the params? I couldn't find it.